Rework openCV test_packages to remove controversial Lena test image [1/2]

ericriff opened the pull request:
See https://github.com/conan-io/conan-center-index/issues/4609

This effort has been splitted in 2 PRs to workaround the CCI limitation of only modifying one recipe per PR. Another PR will fix 3.x.

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

ericriff commented:
Why does 2 version of the same library count as 2 separate libraries? :(

prince-chrismc commented:
it's two different recipes

Croydon commented:
Best case scenario would be a test that does not need a binary file in git

ericLemanissier commented:
Conan-center-index should allow this situation, because it is really no different than modifying a single conanfile which builds several version. In both case there can be no requirement relation between the packages to build, so the order of build does not matter.

mathbunnyru commented on recipes/opencv/2.x/test_package/conanfile.py:
```diff
@@ -14,6 +14,6 @@ def build(self):
+            peppers = os.path.join(self.source_folder, "peppers.jpg")
             res_path = self.deps_cpp_info["opencv"].res_paths[0]
-            self.run("%s %s %s" % (bin_path, res_path,lena), run_environment=True)
+            self.run("%s %s %s" % (bin_path, res_path,peppers), run_environment=True)
```
```suggestion
            self.run("{} {} {}".format(bin_path, res_path, peppers), run_environment=True)
```

mathbunnyru commented on recipes/opencv/3.x/test_package/conanfile.py:
```diff
@@ -14,6 +14,6 @@ def build(self):
+            peppers = os.path.join(self.source_folder, "peppers.jpg")
             res_path = self.deps_cpp_info["opencv"].res_paths[0]
-            self.run("%s %s %s" % (bin_path, res_path,lena), run_environment=True)
+            self.run("%s %s %s" % (bin_path, res_path,peppers), run_environment=True)
```
```suggestion
            self.run("{} {} {}".format(bin_path, res_path, peppers), run_environment=True)
```

mathbunnyru reviewed (dismissed):
Let's update the code a little bit, if you're changing the lines.

ericriff commented:
So whatever approach we take, I have to split this into 2 PRs then?

mathbunnyru commented:
Yep, I think the fastest way to get this merged is to create 2 different PRs.

uilianries reviewed (changes requested):
Please, remove png files and keep test more simple.

ericriff commented:
I reverted the changes on 3.x (will open another PR later), got rid of the peppers image on 2.x and replaced the test_package with one similar to the one on 4.x, which doesn't use an image.

mathbunnyru commented on recipes/opencv/2.x/test_package/test_package.cpp:
```diff
@@ -1,67 +1,164 @@
 
-        Mat faceROI = frame_gray( faces[i] );
-        std::vector<Rect> eyes;
+#define w 400
```
meh

mathbunnyru reviewed (dismissed):
test_package is bigger than usual, but that's fine by me.

ericriff commented:
> test_package is bigger than usual, but that's fine by me.

I know, I usually include a couple of headers and call a function. Just check for include dirs and linking. But I wanted to be consistent with what we have on 4.x.

This is a copy/pase of openCV official code.

mathbunnyru commented:
> > test_package is bigger than usual, but that's fine by me.
> 
> I know, I usually include a couple of headers and call a function. Just check for include dirs and linking. But I wanted to be consistent with what we have on 4.x.
> 
> This is a copy/pase of openCV official code.

Oh, I see.
Yes, having consistent code is a good thing.
Thanks for the reply.

uilianries reviewed (approved):
Great! Thank you!!
